extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultPKISecretEngine.java

```diff
@@ -0,0 +1,218 @@
+package io.quarkus.vault;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+
+import io.quarkus.vault.pki.ConfigCRLOptions;
+import io.quarkus.vault.pki.ConfigURLsOptions;
+import io.quarkus.vault.pki.GenerateCertificateOptions;
+import io.quarkus.vault.pki.GenerateIntermediateCSROptions;
+import io.quarkus.vault.pki.GenerateRootOptions;
+import io.quarkus.vault.pki.GeneratedCertificate;
+import io.quarkus.vault.pki.GeneratedIntermediateCSRResult;
+import io.quarkus.vault.pki.GeneratedRootCertificate;
+import io.quarkus.vault.pki.RoleOptions;
+import io.quarkus.vault.pki.SignIntermediateCAOptions;
+import io.quarkus.vault.pki.SignedCertificate;
+import io.quarkus.vault.pki.TidyOptions;
+
+/**
+ * A service that interacts with Hashicorp's Vault PKI secret engine to issue certificates & manage certificate
+ * authorities.
+ *
+ * @see <a href="https://www.vaultproject.io/docs/secrets/pki">PKI</a>
+ */
+public interface VaultPKISecretEngine {
+
+    /**
+     * Retrieves the engine's PEM encoded CA certificate.
+     * 
+     * @return PEM encoded certificate authority certificate .
+     */
+    String getCertificateAuthority();
+
+    /**
+     * Configures the engine's CA.
+     *
+     * @param pemBundle PEM encoded bundle including the CA, with optional chain, and private key.
+     */
+    void configCertificateAuthority(String pemBundle);
+
+    /**
+     * Configures engine's URLs for issuing certificates, CRL distribution points, and OCSP servers.
+     *
+     * @param options URL options
```

Comment:
be consistent with ending param description with a dot (always or never).

Reply:
👍